src/main/java/org/spongepowered/api/network/ChannelRegistrar.java

```diff
@@ -60,6 +61,62 @@
     ChannelBinding.RawDataChannel createRawChannel(Object plugin, String channel) throws ChannelRegistrationException;
 
     /**
+     * Gets a channel binding if a channel registered by that name exists.
+     *
+     * @param channel The name of the channel
+     * @return The channel if it exists
+     */
+    Optional<ChannelBinding> getChannel(String channel);
+
+    /**
+     * Gets or creates a {@link ChannelBinding.IndexedMessageChannel} by the
+     * given name. If the channel exists and is a indexed message channel, then
+     * it is returned. If the channel is not an indexed message channel, then
+     * {@link Optional#empty()} is returned. Otherwise, a new channel is
+     * created.
+     *
+     * @param plugin The plugin to register the channel if it doesn't exist
+     * @param channel The channel name
+     * @return A new or existing indexed message channel binding
+     * @throws ChannelRegistrationException for same reasons as
+     *         {@link #createChannel}.
+     */
+    default Optional<ChannelBinding.IndexedMessageChannel> getOrCreate(Object plugin, String channel) throws ChannelRegistrationException {
```

Comment:
There's an argument to be made for throwing a different exception at that point, since if the channel is registered but of a different type, then it would indicate some kind of conflict (maybe another plugin using the same channel, maybe an older/newer version which uses raw channels instead), so maybe it's worth throwing a `UnexpectedChannelTypeException` or something which indicates the conflict state?